my_sprites.py

```diff
@@ -1,4 +1,119 @@
+            grid_size: int,
+            boundary_left: int,
+            boundary_right: int,
+            boundary_bottom: int,
+            boundary_top: int,
+            scale=1.0):
+
+        super().__init__(
+            filename=filename,
+            scale=scale,
+            center_x=center_pos[0],
+            center_y=center_pos[1]
+        )
+
+        # hp
+        self._max_hp = max_hp
+        self.cur_hp = max_hp
+
+        self.speed = speed
+
+        # pathfinding
+        self.path = []
+        self.cur_path_position = 0  # which point on the path we are heading for.
+
+        # create our own map of barriers
+        self.barriers = arcade.AStarBarrierList(
+            moving_sprite=self,
+            blocking_sprites=impassables,
+            grid_size=grid_size,
+            left=boundary_left,
+            right=boundary_right,
+            bottom=boundary_bottom,
+            top=boundary_top
+        )
+
+    @property
+    def max_hp(self):
+        return self._max_hp
+
+    def go_to_point(self, target_pos: tuple[int, int]):
+        """
+        calculates a path to the target pos. Sets the sprite's path to this path.
+        If an enemy has a path, it will automatically follow it.
+        If no barrier list is given, use the sprites own barriers.
+        """
+
+        # calculate the path. It will be a list of positions(lists)
+        self.path = arcade.astar_calculate_path(self.position,
+                                                target_pos,
+                                                self.barriers,
+                                                diagonal_movement=True)
+
+        # reset this because we are at the start of a new path
+        self.cur_path_position = 0
+
+    def on_update(self, delta_time: float = 1 / 60):
+
+        # follow the path, if present
+        if self.path:
+
+            # next position to move to
+            dest_pos = self.path[self.cur_path_position]
+
+            # calculate angle to next point
+            angle_to_dest = arcade.get_angle_radians(dest_pos[0], dest_pos[1], self.center_x, self.center_y)
+
+            # calculate distance
+            distance_to_dest = arcade.get_distance(dest_pos[0], dest_pos[1], self.center_x, self.center_y)
+
+            # this is so we don't move too far
+            this_move_length = min(self.speed, distance_to_dest)
+
+            # if we are there, set the next position to move to
+            if distance_to_dest <= self.speed:
+                self.cur_path_position += 1
+
+                # if we are finished with this path, stand still
+                if self.cur_path_position == len(self.path):
+                    self.path = []
+
+            # testing shows that we need to reverse the direction...
+            self.change_x = -math.sin(angle_to_dest) * this_move_length
+            self.change_y = -math.cos(angle_to_dest) * this_move_length
+
+        self.center_x += self.change_x
```

Comment:
Should these not be intented one level so we only update the position if we are not too close to dest?

Reply:
you mean with an _else_?

Reply:
wouldn't it be prettier to set change_x and then update it later?
Im thinking of when we are not moving along a path